executor/partition_table_test.go

```diff
@@ -305,6 +305,65 @@ func (s *partitionTableSuite) TestGlobalStatsAndSQLBinding(c *C) {
 	tk.MustIndexLookup("select * from tlist where a<1")
 }
 
+func createTable4DynamicPruneModeTestWithExpression(tk *testkit.TestKit) {
+	tk.MustExec("create table trange(a int) partition by range(a) (partition p0 values less than(3), partition p1 values less than (5), partition p2 values less than(11));")
+	tk.MustExec("create table thash(a int) partition by hash(a) partitions 4;")
+	tk.MustExec("create table t(a int)")
+	tk.MustExec("insert into trange values(1), (2), (3), (4), (5), (6), (7), (10);")
+	tk.MustExec("insert into thash values(1), (2), (3), (4), (5), (6), (7), (10);")
+	tk.MustExec("insert into t values(1), (2), (3), (4), (5), (6), (7), (10);")
```

Comment:
Please also insert some `NULLs` and duplicated records.

Reply:
`NULL records` is another TiCase. I just add NULL value here and I'll add test cases for NULL value in the later PR.